src/release_artifacts_resources/ios/cdk/cdk/credential_rotation/lambda_functions/credential_rotator.py

```diff
@@ -36,26 +37,65 @@
 CIRCLECI_EXECUTION_ROLE = os.environ.get("CIRCLECI_EXECUTION_ROLE")
 IAM_USERNAME = os.environ.get("IAM_USERNAME")
 GITHUB_PROJECT_PATH = os.environ.get("GITHUB_PROJECT_PATH")
+GITHUB_CREDENTIALS_SECRET = os.environ.get("GITHUB_CREDENTIALS_SECRET")
+
+RELEASE_BUCKET_NAME = os.environ.get("RELEASE_BUCKET_NAME")
+RELEASE_CLOUDFRONT_DISTRIBUTION_ID = os.environ.get("RELEASE_CLOUDFRONT_DISTRIBUTION_ID")
 
 random.seed()
 
 
 def handler(event, context, *, iam=None, sts=None, secretsmanager=None):
     iam_client = iam or boto3.client("iam", region_name=REGION)
     secretsmanager_client = secretsmanager or boto3.client("secretsmanager", region_name=REGION)
-
+    circleci_api_token = get_secret_value(
+        CIRCLECI_CONFIG_SECRET, secretsmanager=secretsmanager_client
+    )
     user_credentials: Tuple[str, str] = ()
     try:
         user_credentials = create_user_credentials(IAM_USERNAME, iam=iam_client)
         wait_for_user_credentials()
         session_credentials = get_session_credentials(user_credentials, sts=sts)
-        circleci_api_token = get_secret_value(
-            CIRCLECI_CONFIG_SECRET, secretsmanager=secretsmanager_client
-        )
         update_env_vars(session_credentials, circleci_api_token, GITHUB_PROJECT_PATH)
     finally:
         if user_credentials:
             iam_client.delete_access_key(UserName=IAM_USERNAME, AccessKeyId=user_credentials[0])
+    update_bucket_cloudfront_info(circleci_api_token)
+    update_github_credentials(secretsmanager_client, circleci_api_token)
```

Comment:
Why aren't these in the `try` block? I think I understand it's because the temporary session credentials have different cleanup steps, but it is a little odd to have the different pattern here. Consider refactoring the session credentials into an `update_session_credentials` helper method, so `handler` is simpler

Reply:
Moved to a separate helper method.